### ISSUE TITLE
Update Suite Design Guide update for xtriggers

### ISSUE THREAD

**Issue by ColemanTom** [cylc/cylc#2909]
I was wondering if the Suite Design Guide could be updated with some details about `xtriggers`. I cannot see them mentioned anywhere in there, and I would personally find it useful to adopt a standard approach to using them from the outset. I don't know what to suggest as I have not used them yet. Or are they still too fresh to include in the design guide?

**Comment by sadielbartholomew** [comment]
I might not be the best one to answer your query, but I few points to note on this (you may be aware already, & if so my apologies):

* Since the SDG has not been updated since March 2017, & there have been a significant number of major changes to both Cylc & Rose since then, I can only imagine there are a number of topics we would want to detail in the next SDG update. However, since we are all largely pre-occupied with a [intensive roadmap towards Cylc 8](https://github.com/cylc/cylc-admin/blob/master/docs/cylc-8-tasks.md), I doubt there will be an update anytime soon. If you want specific design-based advice that is not documented, such as in this case on 'xtriggers', I suggest emailing us (e.g. the [Metomi team](https://github.com/metomi)).
* I am currently in the process of converting the documentation, including the SDG, to be auto-generated via Sphinx instead of sourced from LaTeX. This will mean the SDG source files will change format, likely not too long after the New Year. So be wary that any suggested changes to the documentation may need to be reformatted after the merge of that PR.

**Comment by hjoliver** [comment]
@sadielbartholomew - I agree, we need to update the SDG once cylc-8 is in hand.  I suspect that not many changes in the last 18 months will warrant major changes in the SDG advice, although external triggers is a big one. (However xtriggers are documented fairly extensively in the User Guide - presumably you have seen that, @ColemanTom ? - with the advice for the moment to avoid wholesale replacement of existing clock triggers and inter-suite polling tasks with their new built-in xtrigger counterparts ... which I think gives us a little wriggle-room on the SDG update).

**Comment by ColemanTom** [comment]
@hjoliver yes, I've read the xtriggers section. I was more trying to point out that they are not mentioned in there, and that is a gap in the `design guide`.
